src/main/java/io/confluent/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -103,6 +103,7 @@ public BufferedRecords(
           config.pkMode,
           schemaPair,
           fieldsMetadata,
+          dbStructure.tableDefinition(connection, tableId),
```

Comment:
This is not a safe change to make, because if someone implemented DatabaseDialect and implemented the old method, their implementation would never be called.

Reply:
> This is not a safe change to make, because if someone implemented DatabaseDialect and implemented the old method, their implementation would never be called.

I think now it is safe. dialects that didn't overwrite the overloaded statementBinder in the interface, shall be redirect to the old behavior.